packages/nextjs/components/scaffold-eth/Balance.tsx

```diff
@@ -14,10 +14,10 @@ type BalanceProps = {
 /**
  * Display (ETH & USD) balance of an ETH address.
  */
-export const Balance = ({ address, className = "", usdMode }: BalanceProps) => {
+export const Balance = ({ address, className = "", usdMode = false }: BalanceProps) => {
   const { targetNetwork } = useTargetNetwork();
   const { balance, price, isError, isLoading } = useAccountBalance(address);
-  const [displayUsdMode, setDisplayUsdMode] = useState(price > 0 ? Boolean(usdMode) : false);
+  const [displayUsdMode, setDisplayUsdMode] = useState(Boolean(usdMode));
```

Comment:
It seems like `price` is always `0` on initial load and therefore the `Boolean(usdMode)` was always false as the initial state, no matter what. This caused the `usdMode` flag to not work even when applied, so I removed the `price > 0` logic and it works.

Reply:
This seems like an issue on main SE-2 too. Not sure why it was added in the first place. Will do some tinkering and maybe a PR.

Thanks!

Reply:
Update: Created this PR https://github.com/scaffold-eth/scaffold-eth-2/pull/856

Initially I removed the `price > 0` condition as you did here, but there are two issues with it:
1. If the price fails to load, the usdMode is useless since we don't have and available conversion
2. You can't control the usdMode outside the Balance component (this might not be important here, but it's a good to have in main SE-2).

So I ended up adding a `useEffect`, as we do in `EtherInput`

Reply:
Ok thanks, I'll keep an eye out on the situation there so I can copy the solution if needed

Reply:
Cool! With the current solution the problem is that if the price of ETH fails to load, it will show `$ 0.00` forever. With the condition (+ the missing use effect) it will show the ETH price, which I think is better.